cmd/bb_runner/main.go

```diff
@@ -36,10 +38,38 @@ func main() {
 		log.Fatal("Failed to open build directory: ", err)
 	}
 
-	r := runner.NewLocalRunner(
-		buildDirectory,
-		configuration.BuildDirectoryPath,
-		configuration.SetTmpdirEnvironmentVariable)
+	rootDirectoryOpener := func(inputRootDirectory string) (filesystem.DirectoryCloser, error) {
```

Comment:
It may be wiser to convert this to:
```go
var rootDirectoryOpener runner.RootDirectoryOpener
if configuration.ChrootIntoInputRoot {
   rootDirectoryOpener = func(...) {
       ...
   }
} else {
   rootDirectoryOpener = func(...) {
       ...
   }
}
```
That way the boolean comparison is only performed once during startup.

Another thing that you could consider doing is opening the root directory just once during bb_runner startup. You then need to make sure that this directory is never closed, but you can solve that by calling `filesystem.NopDirectoryCloser()` to give a `filesystem.Directory` a no-op `Close()` operation.